test/integration/link-with-hash/test/index.test.js

```diff
@@ -0,0 +1,53 @@
+/* eslint-env jest */
+
+import { join } from 'path'
+import webdriver from 'next-webdriver'
+import {
+  findPort,
+  launchApp,
+  killApp,
+  nextStart,
+  nextBuild,
+} from 'next-test-utils'
+
+jest.setTimeout(1000 * 60 * 5)
+let app
+let appPort
+const appDir = join(__dirname, '..')
+
+const runTests = () => {
+  it('should not have hydration mis-match for hash link', async () => {
+    const browser = await webdriver(appPort, '/')
+    const browserLogs = await browser.log('browser')
+    let found = false
+    browserLogs.forEach((log) => {
+      if (log.message.includes('Warning: Prop')) {
```

Comment:
Hi, this warning won't show in production mode so we don't need to test that, we could add this check to the client-navigation suite [here](https://github.com/vercel/next.js/blob/598997f038fcbc5345ddb3320194a2a6c741b59a/test/integration/client-navigation/test/index.test.js#L552) using the `/nav/hash-changes` page to keep the test related to that. The check looks good besides that!

Reply:
@ijjk (cc: @timneutkens )
Hi, thanks for the review!
You are right that this warning does not show in production.
I was not sure where to put this test, so I created new one. Adding this check to the client-navigation suite using the` /nav/hash-changes` makes sense to me.
Since this PR is merged, should I make an another PR to apply your suggestion?

Reply:
If you'd like to create another PR updating the test that would be awesome!